doc/guides/weight_normalization.rst

```diff
@@ -0,0 +1,51 @@
+Weight normalization
+====================
+
+Preliminaries
+-------------
+
+Suppose that the incoming synaptic weights of a neuron are given as :math:`w_1, w_2, \ldots, w_n`. A plasticity rule might require that the vector norm :math:`|w|` remains constant. For example, the 1-norm is used in [1]_ [2]_:
+
+.. math::
+
+   |W|_1 = \sum_i W_i
+
+Keeping this norm constant is typically done as an extra step after the main weights plasticity step (e.g. after an STDP weight update). First, the norm is computed, and second, all weights :math:`w_1, \ldots, w_n` are updated according to:
+
+.. math::
+
+   W_i &\leftarrow W_{total} \frac{W_i}{|W|_1}
+
+where :math:`w_{total}` is the target total input weight [2]_.
+
+
+Implementation in NEST
+----------------------
+
+Because of the way that the data structures are arranged in NEST, normalizing the weights is a costly operation (in terms of time spent). One has to iterate over all the neurons, then for each neuron fetch all of its incoming connections, calculate the vector norm and perform the actual normalization, and finally to write back the new weights.
+
+This would look something like:
+
+.. code-block:: python
+
+   def normalize_weights(neuron_gids_to_be_normalized):
```

Comment:
One option here would be to say right from the start that `w_{total}` is assumed to be one or, which I would prefer, to pass `w_total` as an argument to the function such that there is correspondence in code and equation above.